sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfigPropertiesTest.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ExtendedConfigProperties;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfiguration;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FileConfigPropertiesTest {
+
+  private static final String extendedSchema =
+      "file_format: \"0.1\"\n"
+          + "disabled: false\n"
+          + "\n"
+          + "resource:\n"
+          + "  attributes:\n"
+          + "    service.name: !!str \"unknown_service\"\n"
+          + "\n"
+          + "other:\n"
+          + "  str_key: str_value\n"
+          + "  int_key: 1\n"
+          + "  float_key: 1.1\n"
+          + "  bool_key: true\n"
+          + "  str_list_key: [val1, val2]\n"
+          + "  int_list_key: [1, 2]\n"
+          + "  float_list_key: [1.1, 2.2]\n"
+          + "  bool_list_key: [true, false]\n"
+          + "  map_key:\n"
+          + "    str_key1: str_value1\n"
+          + "    int_key1: 2\n"
+          + "    map_key1:\n"
+          + "      str_key2: str_value2\n"
+          + "      int_key2: 3\n"
+          + "  list_key:\n"
+          + "    - str_key1: str_value1\n"
+          + "      int_key1: 2\n"
+          + "      map_key1:\n"
+          + "        str_key2: str_value2\n"
+          + "        int_key2: 3\n"
+          + "    - str_key1: str_value1\n"
+          + "      int_key1: 2";
```

Comment:
Check out this test for examples of how to use the API to read complex types defined in a configuration file.

Notably, I've added support for dot notation to allow existing system property / environment variable configuration to be translated to a hierarchical representation in YAML.

For example, the property `otel.javaagent.debug` would be represented in YAML with:
```
otel:
  javaagent:
    debug: true
```
And you could access it using `configProperties.getBoolean("otel.javaagent.debug")`.